Source/WebCore/rendering/style/RenderStyle.cpp

```diff
@@ -1312,27 +1312,31 @@ inline static bool changedCustomPaintWatchedProperty(const RenderStyle& a, const
 
         for (auto& watchPropertiesMap : { propertiesA, propertiesB }) {
             for (auto& name : watchPropertiesMap) {
-                RefPtr<const CSSValue> valueA;
-                RefPtr<const CSSValue> valueB;
                 if (isCustomPropertyName(name)) {
-                    valueA = a.customPropertyValue(name);
-                    valueB = b.customPropertyValue(name);
+                    auto valueA = a.customPropertyValue(name);
+                    auto valueB = b.customPropertyValue(name);
+
+                    if ((valueA && !valueB) || (!valueA && valueB))
+                        return true;
+                    if (!valueA)
+                        continue;
+                    if (*valueA != *valueB)
+                        return true;
```

Comment:
```suggestion
                    if (valueA != valueB && (!valueA || !valueB || *valueA != *valueB))
                        return true;
```
Or we can make a helper function that handles the null checking and write it as just a single function call.